java-monitoring/run_protoc.sh

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+
+set -e
+
+LIBRARY_GEN_OUT=$(pwd)/../library_gen_out
+
+mkdir -p "${LIBRARY_GEN_OUT}"/google/monitoring/v3
+
+##################### Section 0 #####################
+# prepare tooling
+#####################################################
+# proto files from googleapis repository
+REPO_ROOT=$(pwd)/..
+cd "${REPO_ROOT}"
+
+if [ ! -d googleapis ]; then
+  git clone https://github.com/googleapis/googleapis.git
+fi
+
+GOOGLEAPIS_ROOT=${REPO_ROOT}/googleapis
+cd "${GOOGLEAPIS_ROOT}"
+git checkout 53a0be29c4a95a1d3b4c0d3a7a2ac8b52af2a3c0
+PROTO_FILES=$(ls google/monitoring/v3/*.proto)
+# pull proto files and protoc from protobuf repository
+# maven central doesn't have proto files
+cd "${REPO_ROOT}"
+curl -LJ -o protobuf.zip https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip
+unzip -o -q protobuf.zip -d protobuf/
+cp -r protobuf/include/google "${GOOGLEAPIS_ROOT}"
+PROTOC_ROOT=${REPO_ROOT}/protobuf/bin
+echo "protoc version: $("${PROTOC_ROOT}"/protoc --version)"
+# pull protoc-gen-grpc-java plugin from maven central
+cd "${LIBRARY_GEN_OUT}"
+curl -LJ -o protoc-gen-grpc-java https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.54.1/protoc-gen-grpc-java-1.54.1-linux-x86_64.exe
+chmod +x protoc-gen-grpc-java
+# gapic-generator-java
+curl -LJ -o gapic-generator-java.jar https://repo1.maven.org/maven2/com/google/api/gapic-generator-java/2.19.0/gapic-generator-java-2.19.0.jar
+##################### Section 1 #####################
+# generate grpc-*/
+#####################################################
+cd "${GOOGLEAPIS_ROOT}"
+"${PROTOC_ROOT}"/protoc "--plugin=protoc-gen-rpc-plugin=${LIBRARY_GEN_OUT}/protoc-gen-grpc-java" \
+"--rpc-plugin_out=:${LIBRARY_GEN_OUT}/google/monitoring/v3/monitoring_java_grpc-proto-gensrc.jar" \
```

Comment:
Can we try to extract "monitoring" and "v3" into variables, so that the script would be more generic?